invenio_communities/assets/semantic-ui/js/invenio_communities/community/components/communities-carousel.js

```diff
@@ -0,0 +1,230 @@
+    const { title, animationSpeed } = this.props;
+
+    return data.hits.length ? (
+      <Container fluid className="carousel rel-pt-2 rel-pb-2 ml-0-mobile mr-0-mobile">
+        {title ? (
+          <Container className="rel-mb-1">
+            <Header as="h2">{title}</Header>
+          </Container>
+          ) : ""
+        }
+
+        <Grid
+          container
+          onFocus={this.stopCarousel}
+          onBlur={this.startCarousel}
+        >
+          <Grid.Column
+            width="2"
+            className="pr-0"
+            verticalAlign="middle"
+            textAlign="left"
+          >
+            <Icon
+              className="carousel-arrow"
+              inverted
+              role="button"
+              name="angle left"
+              size="huge"
+              aria-label={i18next.t("Previous slide")}
+              onClick={() => this.runCarousel(activeIndex - 1)}
+              onKeyDown={(event) => event.key === "Enter" && this.runCarousel(activeIndex - 1)}
+              tabIndex="0"
+            />
+          </Grid.Column>
+
+          <Grid.Column width="12" className="flex align-items-center">
+            <Transition.Group
+              as={Item.Group}
+              className="flex align-items-center justify-center"
+              duration={animationSpeed}
+              animation={`carousel-slide ${animationDirection}`}
+              directional={true}
+            >
+              {this.renderItem()}
+            </Transition.Group>
+          </Grid.Column>
+
+          <Grid.Column
+            width="2"
+            className="pl-0"
+            verticalAlign="middle"
+            textAlign="right"
+          >
+            <Icon
+              className="carousel-arrow"
+              inverted
+              role="button"
+              name="angle right"
+              size="huge"
+              aria-label={i18next.t("Next slide")}
+              onClick={() => this.runCarousel(activeIndex + 1)}
+              onKeyDown={(event) => event.key === "Enter" && this.runCarousel(activeIndex + 1)}
+              tabIndex="0"
+            />
+          </Grid.Column>
+        </Grid>
+      </Container>
+    ) : "";
+  }
+}
+
+CommunitiesCarousel.propTypes = {
+  title: PropTypes.string.isRequired,
+  fetchUrl: PropTypes.string.isRequired,
+  intervalDelay: PropTypes.number.isRequired,
+  animationSpeed: PropTypes.number.isRequired,
+  defaultLogo: PropTypes.string.isRequired,
+};
+
+const communitiesCarouselContainer = document.getElementById("communities-carousel");
+
+const title = communitiesCarouselContainer.dataset.title;
+const fetchUrl = communitiesCarouselContainer.dataset.fetchUrl;
+const intervalDelay = parseInt(communitiesCarouselContainer.dataset.intervalDelay);
+const animationSpeed = parseInt(communitiesCarouselContainer.dataset.animationSpeed);
+const defaultLogo = communitiesCarouselContainer.dataset.defaultLogo;
+
+communitiesCarouselContainer && ReactDOM.render(
```

Comment:
I would move the render part in an `index.js` file. Like that when we import the component from somewhere we do all these extra redundant work with the dom queries...